pandas/core/resample.py

```diff
@@ -1388,10 +1388,16 @@ def __init__(
             if label is None:
                 label = "right"
         else:
-            if closed is None:
-                closed = "left"
-            if label is None:
-                label = "left"
+            if origin in ["end", "end_day"]:
```

Comment:
can you give some comments on the rationale here (essentially what you described in the docs), for future readers